src/modals/SharePoll.jsx

```diff
@@ -1,16 +1,103 @@
 import React from 'react';
 import { useLocation } from 'react-router-dom';
+import styled from "styled-components";
+import Button from "../components/Button";
+import ShareIcon from 'assets/ShareIcon.png';
 
+const ModalIconField = styled.div`
+  height: 50px;
+  width: 50px;
+  background: #F5DFBD 0% 0% no-repeat padding-box;
+  border-radius: 50%;
+  display: inline-block;
+  align-self: center;
+`;
+
+const ModalHeader = styled.h5`
+  text-align: center;
+  letter-spacing: 0px;
+  color: #000000;
+  opacity: 1;
+  font-weight: normal;
+  font-size: 30px;
+  padding-top: 20px;
+  padding-bottom: 22px;
+`;
+
+const ModalIcon = styled.img`
+  height: 32px;
+  width: 36px;
+  position: relative;
+  left: 7px;
+  top: 9px;
+`;
+
+const ModalContent = styled.p`
+    text-align: left;
+    letter-spacing: 0px;
+    color: #343A40;
+    opacity: 1;
+    font-weight: normal;
+    font-size: 16px;
+    padding-bottom: 56px;
+`;
+
+const ModalFooter = styled.footer`
+    text-align: left;
+    letter-spacing: 0px;
+    color: #343A40;
+    opacity: 1;
+    font-weight: normal;
+    font-size: 14px;
+    padding-top: 56px;
+    padding-bottom: 10px;
+`;
+
+const URLfield = styled.div`
+    background: #FFFFFF 0% 0% no-repeat padding-box;
+    border: 2px solid #E8EBEE;
+    border-radius: 10px;
+    position: relative;
+    padding-top: 6px;
+    padding-bottom: 3px; 
+    padding-left: 20px;
+    left: 5%;
+    width: 90%;
+    opacity: 1;
+    font-size: 18px;
+    letter-spacing: 0px;
+    color: #343A40;
+    opacity: 1;
+`;
 
 function SharePoll(props) {
-  let location = useLocation();
-
-  return (
-    <>
-      <div>Share Poll Modal</div>
-      <p>{location.pathname}</p>
-    </>
-  );
+    const {closeModal} = props;
+    let location = useLocation();
+
+    return (
+        <>
+            <ModalIconField>
+                <ModalIcon src={ShareIcon}></ModalIcon>
+            </ModalIconField>
+            <ModalHeader>
+                Udostępnij ankietę innym
+            </ModalHeader>
+            <ModalContent>
+                Aby udostępnić ankietę po prostu skopiuj i prześlij ten link:
+            </ModalContent>
+            <URLfield>{location.pathname}</URLfield>
+            <ModalFooter>
+                Psst! Nie zapomnij przekazać także kodu PIN: 2137
+            </ModalFooter>
```

Comment:
Tutaj trzy rzeczy. Po pierwsze nazewnictwo komponentów co w sumie jest preferencją ale... nie musisz przed każdym z ostylowanych elementów pisać że jest to np. `ModalHeader`, `ModalContent` itp. Można po prostu napisać `Header` i z kontekstu samo przez się tłumaczy że jest to header modalu. Jak w innym miejscu będzie inny ostylowany `Header` to nam to nie przeszkadza. Po drugie puste elementy można pisać w skróconej wersji np. `<ModalIcon />` tak samo jak w HTML jak są elementy `<img />` czy `<br />`. Po trzecie (co też jest w sumie kwestią preferencji) nie trzeba tekstu umieszczać w każdej nowej linijce. Po tych trzech zmianach robi nam się zdecydowanie krótszy kod a równie czytelny:
```HTML
      <IconField><ModalIcon src={ShareIcon}/></IconField>
      <Header>Udostępnij ankietę innym</Header>
      <Content>Aby udostępnić ankietę po prostu skopiuj i prześlij ten link:</Content>
      <URLfield>{location.pathname}</URLfield>
      <Footer>Psst! Nie zapomnij przekazać także kodu PIN: 2137</Footer>
```